container/go/cmd/create_image_config/createImageConfig.go

```diff
@@ -0,0 +1,413 @@
+// Copyright 2016 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+////////////////////////////////////////////////
+// This package manipulates v2.2 image configuration metadata.
+// It writes out both a config file and a manifest for the v2.2 image.
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/pkg/errors"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	// "github.com/google/go-containerregistry/pkg/v1/mutate"
+)
+
+var (
+	baseConfig       = flag.String("baseConfig", "", "The parent image.")
+	baseManifest     = flag.String("baseManifest", "", "The parent image manifest.")
+	outputConfig     = flag.String("outputConfig", "", "The output file to generate.")
+	outputManifest   = flag.String("outputManifest", "", "The manifest output file to generate.")
+	creationTime     = flag.String("creationTime", "", "The creation timestamp. Acceptable formats: Integer or floating point seconds since Unix Epoch, RFC, 3339 date/time.")
+	user             = flag.String("user", "", "The username to run the commands under.")
+	workdir          = flag.String("workdir", "", "Set the working directory of the layer.")
+	nullEntryPoint   = flag.String("nullEntryPoint", "false", "If true, Entrypoint will be set to null.")
+	nullCmd          = flag.String("nullCmd", "false", "If true, Cmd will be set to null.")
+	operatingSystem  = flag.String("operatingSystem", "linux", "Operating system to create docker image for distro specified.")
+	labelsArray      arrayFlags
+	ports            arrayFlags
+	volumes          arrayFlags
+	entrypointPrefix arrayFlags
+	env              arrayFlags
+	command          arrayFlags
+	entrypoint       arrayFlags
+	layer            arrayFlags
+	stampInfoFile    arrayFlags
+	// empty file sha256 hex
+	b         = sha256.Sum256([]byte(""))
+	emptyFile = hex.EncodeToString(b[:])
+)
+
+// default architecture type based on legacy create_image_config.py
+const processorArchitecture = "amd64"
+
+// unix epoch 0, representation in 32 bits
+const defaultTimestamp = "1970-01-01T00:00:00Z"
+
+// arrayFlags is a type that can handle multiple flags of the same name.
```

Comment:
remove `is a type` which is evident from the declaration syntax.

Also, suggesting saying `that can be used to store multiple flags ...` but this is just a nit. "handle" is too generic a word for this purpose imo.